analyzers/src/SonarAnalyzer.CSharp/Helpers/CSharpMethodParameterLookup.cs

```diff
@@ -23,22 +23,22 @@
 
 namespace SonarAnalyzer.Helpers
 {
-    internal class CSharpMethodParameterLookup : AbstractMethodParameterLookup<ArgumentSyntax>
+    internal class CSharpMethodParameterLookup : MethodParameterLookupBase<ArgumentSyntax>
     {
-        public CSharpMethodParameterLookup(InvocationExpressionSyntax invocation, SemanticModel semanticModel) :
-            this(invocation.ArgumentList, semanticModel)
-        {
-        }
+        public CSharpMethodParameterLookup(InvocationExpressionSyntax invocation, SemanticModel semanticModel)
+            : this(invocation.ArgumentList, semanticModel) { }
+
+        public CSharpMethodParameterLookup(InvocationExpressionSyntax invocation, IMethodSymbol methodSymbol)
+            : this(invocation.ArgumentList, methodSymbol) { }
 
         public CSharpMethodParameterLookup(ArgumentListSyntax argumentList, SemanticModel semanticModel)
-            : base(argumentList?.Arguments, argumentList == null ? null : semanticModel.GetSymbolInfo(argumentList.Parent).Symbol as IMethodSymbol)
-        {
-        }
+            : base(argumentList?.Arguments, argumentList == null ? null : semanticModel.GetSymbolInfo(argumentList.Parent).Symbol as IMethodSymbol) { }
 
         public CSharpMethodParameterLookup(ArgumentListSyntax argumentList, IMethodSymbol methodSymbol)
-            : base(argumentList?.Arguments, methodSymbol)
-        {
-        }
+            : base(argumentList?.Arguments, methodSymbol) { }
```

Comment:
constructor can be single line

Reply:
I intentionally put them on a separate line, as each of them contains specific logic. The longest one would produce too long line anyway and the file structure is regular like this.